src/core/format/KeyGenerator.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2019 NEM
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { UInt64 } from '../../model/UInt64';
+import { sha3_256 } from 'js-sha3';
+
+export class KeyGenerator {
+    /**
+     * Generate UInt64 from a string
+     * @param {string} input Input string
+     * @returns {UInt64} Deterministic uint64 value for the given string
+     */
+    public static fromString(input: string): UInt64 {
+        if (input.length === 0) {
+            throw Error(`Input must not be empty`);
+        }
+        if (input.length > 1024) {
+            throw Error(`Input exceeds 1024 characters (has ${input.length})`);
+        }
+        const hex = sha3_256(input)
+        return UInt64.fromHex(hex.substr(0, 16))
```

Comment:
if you check the definition of UInt64, it uses 2 Uint32 (low / high). I would convert the hash buffer array to uint32 array and return the new Uint64( ['low', 'high']). Not quite sure about the substring here.

I would do something like
```
const result = new Uint32Array(hex.arrayBuffer());
return new UInt64([result[0], (result[1] | 0x80000000) >>> 0]);
```